x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/LdapSessionFactoryTests.java

```diff
@@ -321,7 +320,17 @@ public void testSslTrustIsReloaded() throws Exception {
 
         try (ResourceWatcherService resourceWatcher = new ResourceWatcherService(settings, threadPool)) {
             new SSLConfigurationReloader(resourceWatcher, SSLService.getSSLConfigurations(environment).values()).setSSLService(sslService);
+
+            final FileTime oldModifiedTime = Files.getLastModifiedTime(ldapCaPath);
             Files.copy(fakeCa, ldapCaPath, StandardCopyOption.REPLACE_EXISTING);
+
+            // Force the modified file to have a different modified time
+            // Depending on the granularity of the filesystem it could otherwise be possible the newly copied file looks identical to the
+            // old file (certificates commonly have the same file size)
+            final FileTime newModifiedTime = Files.getLastModifiedTime(ldapCaPath);
+            if (oldModifiedTime.equals(newModifiedTime)) {
```

Comment:
Nit: Why bother with the `if`? The comment makes it clear why this is here, and it shouldn't make a difference to just always do it.

Reply:
I went back and forth on this. 
Part of it reflects the process of getting to the point where I decided that changing the timestamp was the right fix.
It my head, only changing the file if it really needed to be changed felt like the right thing to do, but today I feel the opposite way. I'm going to switch it again. We're explicitly trying to force a resource change, we may as well force a timestamp change all the time.